src/network/networkd-dhcp4.c

```diff
@@ -612,6 +636,11 @@ static int dhcp4_update_address(Link *link,
         addr->prefixlen = prefixlen;
         addr->broadcast.s_addr = address->s_addr | ~netmask->s_addr;
 
+        /* If DHCP route is set then stop creating prefix route. Instead add
+           it later on that table */
+        if (link->network->dhcp_route_table != RT_TABLE_MAIN)
+                addr->flags |= IFA_F_NOPREFIXROUTE;
```

Comment:
This should also look at `link->network->dhcp_route_table_set`, see @yuwata's patch.

I merged the other PR, because it seems to be more carefully written (and CI passes).